### ISSUE TITLE
Need to update to use QT6 to work in OBS 28

### ISSUE THREAD

**Issue by paulpv** [Palakis/obs-ndi#745]
OBS 28 Beta 1 is officially released
https://github.com/obsproject/obs-studio/releases/tag/28.0.0-beta1

> 28.0: Updating to Qt 6
> As of May 2020, version 5 of the Qt user interface toolkit ceased public development and is no longer being updated. Since then, our need for a more up-to-date version has been increasing as time goes by. With version 28.0, we felt it was important for us to update our user interface toolkit to the latest version; not just to gain access to the latest features, but also for the latest bug fixes and improved support for newer operating systems and architectures such as Windows 11 and Apple Silicon. Unfortunately, this also means that certain plugins that depend on version 5 of Qt will no longer work when you upgrade to version 28.0 and will need to be updated.
> 
> During this beta period of OBS 28.0, many of those plugins which rely on Qt version 5 may not yet be updated yet, and we'll be working with third-party plugin developers in order to get them updated before the final release deploys.
> 
> Additionally, Qt 6 has dropped support for Windows 7 & 8, macOS 10.13 & 10.14, Ubuntu 18.04 and all 32-bit operating systems. As such, OBS will no longer be supported on these platforms.
> 
> For more details, please refer to our Discussion Post: https://github.com/obsproject/obs-studio/discussions/6481.

So, this awesome plugin that I use all of the time needs to be updated.

NDI 5.5 SDK also released just a few days ago:
https://www.ndi.tv/sdk/

So, it may be a great time to just refresh this bad boy to also fully support `NDI|HX` or any other newer NDI relevant features!
Maybe setting NDI group id, etc...

@Palakis Interested? I'll be looking in to this either way...

**Comment by fgolman** [comment]
+1 especially on updating NDI!

**Comment by tt2468** [comment]
I know it's been a while since I've touched the `rewrite` branch. I'm slating to update the CMake there and flesh out the rest of the rewrite by the time 28 releases for real.

**Comment by JeffKnight** [comment]
Yes, I use this to have our Proclaim software for screens at church to pass to the OBS stream. I don't want to have to tell people "don't hit update" when they stream services... all it takes is 1 person to be "too helpful" and the thing breaks, so please update!

**Comment by bpmacedo** [comment]
+1!!! Thx 4 the support

**Comment by krystofkrticka** [comment]
Well this was faster than I was expecting OBS 28, already got first release candidate version.

**Comment by tt2468** [comment]
> I know it's been a while since I've touched the `rewrite` branch. I'm slating to update the CMake there and flesh out the rest of the rewrite by the time 28 releases for real.

![20220819_214507.JPG](https://user-images.githubusercontent.com/28720189/185729305-60e538e5-f19f-4721-be3a-e3158dd1657f.JPG)

**Comment by bpmacedo** [comment]
> I know it's been a while since I've touched the `rewrite` branch. I'm slating to update the CMake there and flesh out the rest of the rewrite by the time 28 releases for real.

Great!!! Thank you!

![Screenshot_2022-08-20-01-59-58-818_com android chrome~2](https://user-images.githubusercontent.com/9447019/185729764-fc80dabd-c07f-4246-b851-1a1a29912f9d.jpg)

**Comment by tt2468** [comment]
Hoping to get in a lot of functionality while I'm at it, including aux output capabilities.

**Comment by nacalai** [comment]
+1 on this :) Hoping a brand new plugin with some new neat features. Official release of v28 is closing real fast

**Comment by Riotline** [comment]
> Hoping to get in a lot of functionality while I'm at it, including aux output capabilities.

@tt2468 
Don't know if this would be the right issue thread to add some issues from the rewrite branch but just noticed that NDI output with more than a mono audio channel causes those additional channels to be distorted. 

Can confirm that this only occurred through the obs-ndi plugin and not from NDI as I verified the output was fine through the studio monitor ndi tool.

https://user-images.githubusercontent.com/16257629/186822865-22f65755-4563-45ca-811b-0510674cf7c1.mp4

**Comment by paulpv** [comment]
@tt2468 Love seeing and watching the progress! :) If there is anything that I or anyone can do to help, please post here or PM `pvhaus#7593` on Discord.

**Comment by paulpv** [comment]
@Riotline Can you open a separate issue on what you are seeing?
Please include as much detail as you can think to help debug the issue.

Is it from the standard global NDI output, or is that from a filter on a specific source?
If a filter,  `Dedicated NDI™ output (Audio Only)` or `Dedicated NDI Output`?
What OS?

I use the standard global NDI output as well as a `Dedicated NDI™ output (Audio Only)` all of the time on Windows 10 and have ever seen this.

**Comment by PatchworkPirate** [comment]
Also huge +1 and support on the rewrite to bring in NDI 5.5 as well as OBS 28 support. I use this in broadcast production workflows from time to time and it's an absolute lifesaver!

**Comment by Riotline** [comment]
> OBS 28 is now final. Really need NDI to work with it :)

If you really need it now, you can download the rewrite from the rewrite branch's action artifacts. Note will be little support and bugs. It is currently still being changed but should be progressing.

**Comment by nacalai** [comment]
> Hoping to get in a lot of functionality while I'm at it, including aux output capabilities.

Any timeline? :) We'd love to upgrade to v28 in our production, but is 100% dependent on NDI so just waiting for that. No pressure, just eager :D

**Comment by GibStorm** [comment]
> I don't really see the installer I need under the rewrite branch...?

**Comment by tt2468** [comment]
Still been making progress privately. Will flesh out implementation for program and preview output and fix a few bugs and post an alpha release to tie people over in the short-term.

**Comment by GibStorm** [comment]
> Still been making progress privately. Will flesh out implementation for program and preview output and fix a few bugs and post an alpha release to tie people over in the short-term.

Awesome, thanks.

**Comment by paulpv** [comment]
@tt2468 Let us know if you need any help!

**Comment by tt2468** [comment]
A much better alpha build is now available: https://github.com/Palakis/obs-ndi/actions/runs/2977380826

The only thing it's currently technically missing is the NDI filter. I have some mixed thoughts on it, so I might as well ask the audience.

The NDI filter is a bit of a hack of OBS. It puts a notable strain on the main render thread, and generally does a lot of bad things that can cause problems to propagate to other parts of OBS. This mainly comes as a result of video sending.

My idea is to implement the aux outputs system, allowing you to manage a list of aux outputs via a centralized UI. You select your input/scene, then **select an optional audio track** to use. There is not an effective way to get the raw, unprocessed feed of an OBS source right now that also includes correct timing information. The closest thing we have is audio tracks. I personally feel this functionality would help people more than it would hurt.

With that in mind, I'm interested to hear any notable use-cases where this would not be an effective replacement. With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.

**Comment by Riotline** [comment]
> My idea is to implement the aux outputs system, allowing you to manage a list of aux outputs via a centralized UI.

@tt2468 afaik, OBS 28 includes the PR for ability to have additional video mixes/buses. This could be used for the filter replacement and reduce main render thread strain. I believe they, as an initial implementation, made virtual camera an adjustable video mix

**Comment by tt2468** [comment]
@Riotline Correct, that is what the preview output in the alpha release above is taking advantage of. It only provides that functionality for video, not audio.

**Comment by timokorkalainen** [comment]
Audio mixes would be perfect for my use cases and in fact something I've been hoping for.

Separate aux outputs (instead of filters all over the place) also clears up the output options, so all in all sounds a very good approach.

**Comment by Riotline** [comment]
> @Riotline Correct, that is what the preview output in the alpha release above is taking advantage of. It only provides that functionality for video, not audio.

@tt2468 is audio not supported with the aux outputs? One use case that I personally would sorely adore is to be able to create mix-minus-esque video/audio mix for hosts and transmit it back to them. This would also allow for more applications to see a better reason to implement NDI-in. (Ie teams, zoom)

**Comment by DerThaller** [comment]
> A much better alpha build is now available: https://github.com/Palakis/obs-ndi/actions/runs/2977380826
> 
> The only thing it's currently technically missing is the NDI filter. I have some mixed thoughts on it, so I might as well ask the audience.
> 
> The NDI filter is a bit of a hack of OBS. It puts a notable strain on the main render thread, and generally does a lot of bad things that can cause problems to propagate to other parts of OBS. This mainly comes as a result of video sending.
> 
> My idea is to implement the aux outputs system, allowing you to manage a list of aux outputs via a centralized UI. You select your input/scene, then **select an optional audio track** to use. There is not an effective way to get the raw, unprocessed feed of an OBS source right now that also includes correct timing information. The closest thing we have is audio tracks. I personally feel this functionality would help people more than it would hurt.
> 
> With that in mind, I'm interested to hear any notable use-cases where this would not be an effective replacement. With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.

Hey, how can i Download it? There is no Package or am i just Blind?

**Comment by fgolman** [comment]
I use the dedicated NDI output filter so I can create an additional submix that I use to over lay a clock just for Zoom input, separate from my main output that does not have the clock overlay.

Please continue to provide both the audio and video filters.



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by tt2468** [comment]
> I use the dedicated NDI output filter so I can create an additional submix that I use to over lay a clock just for Zoom input, separate from my main output that does not have the clock overlay.

It sounds like you'd be able to do this just fine using the new method I described.

**Comment by GibStorm** [comment]
Alpha build doesn't work for me. Enabling the Program Output and clicking "OK" doesn't result in network usage at all, it seems. I can find the stream on my other PC in the pulldown menu, but no video or audio is sent across.

**Comment by krystofkrticka** [comment]
I also had few issues but that's expected from alpha version. Already made issue for two potential bugs I found.

**Comment by paulpv** [comment]
I use the Dedicated NDI Audio Output filter all of the time.
It is in fact a requirement in my setup.

I do multi-point streaming, having multiple NDI sources, some local on my network, some coming in over an NDI Bridge.
https://youtu.be/DMyH5HwquX8?t=861

Each scene shows a main/highlight NDI A/V game capture and smaller non-main NDI A/Vs game capture.  
Each endpoint's microphone capture adds a Dedicated NDI Audio Output filter (DNAOf).

All of the endpoint microphone DNAOfs are used in a collection of Compression Side-Chain filters to attenuate all the non-main NDI audios.
(I am even experimenting with the Advanced Scene Switcher plugin to auto-change the scene based on the loudest endpoint microphone DNAOf...but I also have a self-hosted Discord Bot that much more reliably changes the scene based on a voice channel's PTT active talker).

Without a DNAOf the my stream's audio can turn in to a cocktail party where no one watching the stream can make out anyone talking:
https://youtu.be/KuIheGaiFLM?t=123

**Comment by GibStorm** [comment]
> Alpha build doesn't work for me. Enabling the Program Output and clicking "OK" doesn't result in network usage at all, it seems. I can find the stream on my other PC in the pulldown menu, but no video or audio is sent across.

Got it working now. Mistakenly thought the new plugin would just use the old sources. Added in the new NDI5 sources, and everything worked like a charm :)

**Comment by nacalai** [comment]
Trying out the alpha. Seems pretty neat! No problems so far.

Not sure if this should be its own issue but I'll just continue in this thread with a feature request: In the NDI Source properties window, have a "Rename source" button that will rename the source to the NDI source name, or maybe just have it do that automatically or something. That way I don't have to go around renaming all my ndi sources.

Also, I watched eposvox video about obs ndi and that he knows someone who is "taking over" dev of the plugin. Is this you @tt2468  or..? just curious :)

**Comment by OGNetheri** [comment]
+1 for keeping the Dedicated NDI Audio Input filter in. It makes the cross sending of source audios between game and stream-pc easy and certain sources can be only monitored through this method on game-pc without 2 e.g. same websources used and with one pair of headsets (and those websources make duplicates on stream, if they are websourced on both comps so that the sound is heard on gaming-pc headset, if the source is made non seen on game-pc, also the monitoring won't work of course).  

Also, as a simple use case but qof for streamer, is using OBS Soundboard plugin on both pc's while taking streampc's soundboards sounds using DNAOf to be only monitored on game-pc's OBS gives not only at least 20 easy sound to be used through hotkeying them to game-, and stream-pc's numpad but also both obs's SB volume managed with only gaming pc's headphones. Basically this same use case carries to any other sources which would be easiest to be monitored through DNAOf on gaming-pc. 

And the reason to make all possible to stream-pc on sounds is obvious for me: mute on DNAOf'd sources on game-pc won't carry to stream-pc, so it's only logical to keep all non-hardware muted sound sources on stream-pc (where they can truly be muted) only and just monitor them on gaming-pc through DNAOf with one set of headphones. 

Also as a use case to keep DNAOf in for audio is that I think the individual application audio capture (either plugin or the new native beta) is easier to manage just as a one audio output which is sent through dedicated ndi audio output filter.

I bet there's more which speaks behalf of keeping the DNAOf in but these were the few which came to mind on the spot.

**Comment by jcoconnor** [comment]
@Palakis Any update on when a new version of ```obs-ndi``` compatible with ```OBS 28.0``` is coming?

**Comment by nacalai** [comment]
> @Palakis Any update on when a new version of `obs-ndi` compatible with `OBS 28.0` is coming?

Well following this thread you can see @tt2468 already pushed an alpha you can test out

**Comment by ionseon** [comment]
When i copy/paste the obs_ndi.dll and obd_ndi.pdb files into the plugins folder i got an "Plugin.Load.LibError.Message.Windows" Error Message does anyone know how to fix this please?

**Comment by nacalai** [comment]
> When i copy/paste the obs_ndi.dll and obd_ndi.pdb files into the plugins folder i got an "Plugin.Load.LibError.Message.Windows" Error Message when opening OBS, does anyone know how to fix this please? 

Why are you not copy/pasting the full folder structure?

**Comment by ErAzOr2k** [comment]
Any Download link for libndi5 Linux x64?

**Comment by flamingghost** [comment]
Put it through its paces, working well on w10 64bit with obs 28.0.1 @tt2468

**Comment by ionseon** [comment]
> > When i copy/paste the obs_ndi.dll and obd_ndi.pdb files into the plugins folder i got an "Plugin.Load.LibError.Message.Windows" Error Message when opening OBS, does anyone know how to fix this please?
> 
> Why are you not copy/pasting the full folder structure?

Thank you very much, it works now

**Comment by rorj** [comment]
I've been testing [build](https://github.com/Palakis/obs-ndi/actions/runs/2977380826) mentioned in this issue during the weekend. Noticed huge delay building up from the sending side (Windows 11/NDI Runtime 5.0 → macOS 12.15.1). Solved it by installing NDI Tools 5.5 on PC and using included Screen Capture (HX version). Obviously not ideal, since it only supports streaming full desktop, but otherwise works really well, no audio issues yet.

**Comment by OGNetheri** [comment]
At least for me, this unofficial fork works as the plugin worked on V27: https://github.com/dev47apps/obs-ndi/releases/tag/4.9.1

Oe at least the filters work, haven't got the chance to run stream to see does the syncing work as before.

**Comment by jcoconnor** [comment]
> > @Palakis Any update on when a new version of `obs-ndi` compatible with `OBS 28.0` is coming?
> 
> Well following this thread you can see @tt2468 already pushed an alpha you can test out

Thanks - have downloaded and it works good for my purposes at the moment.

**Comment by flexiondotorg** [comment]
> A much better alpha build is now available: https://github.com/Palakis/obs-ndi/actions/runs/2977380826

Thanks for this @tt2468 :+1: I've been testing on Ubuntu 22.04 and noticed that the `obs-ndi-5.0.0-linux-x86_64.deb` from the artefacts above is linked against Qt6 but the `debian/control` file installs Qt5 packages.

To remedy this change `debian/control` as follows:

 - From: `Depends: obs-studio (>= 27.0.0), libqt5core5a (>= 5.9.0~beta), libqt5gui5 (>= 5.3.0), libqt5widgets5 (>= 5.7.0)`

 - To: `Depends: obs-studio (>= 28.0.0), libqt6core6 (>= 6.2.0), libqt6gui6 (>= 6.1.2), libqt6widgets6 (>= 6.1.2)`

**Comment by PinballWizard4** [comment]
> I would very likely only add audio output, and skip video output altogether.

Hi, just registered to comment.
I currently use the filter to transfer two separate video sources from one room (Laptop) to the other room, where my PC is.

If you remove the filter or switch it to audio only, I will only be able to transfer a whole scene / the preview panel, is that right?
That would be really unfortunate, it was such a simple and helpful solution.

Thank you for all your work anyway, even if I probably can't use it in my setup anymore - and sorry if I misunderstood.

Is there any other way or tool to transfer two separate video sources with just a few clicks?
AFAIK vdo ninja is only online and not LAN, and that's not an option for me.

**Comment by tt2468** [comment]
> If you remove the filter or switch it to audio only, I will only be able to transfer a whole scene / the preview panel, is that right?
Inputs (Conventionally called sources), Scenes, Preview, and Program are the plan.

**Comment by PinballWizard4** [comment]
> Inputs (Conventionally called sources), Scenes, Preview, and Program are the plan.

So I absolutely misunderstood. I'm sorry, and thank you for the fast reply. <3
You're awesome!

**Comment by MorningStarGG** [comment]
Not sure why but the alpha version seems to freeze up when streaming and playing a game on my main machine being sent using Screen Capture HX on a RTX 2060. I know it's alpha but figured I'd chime in and see if anyone else has had the issue or if it's known. The video feed just seems to freeze and then OBS locks up and has to be forced close and restarted to get the feed back.

**Comment by PoizenJam** [comment]
> With that in mind, I**'m interested to hear any notable use-cases where this would not be an effective replacement.** With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.

Potential use case. Myself and my partner co-stream. Using NDI filters, we are able to share the raw streams of several video and audio capture sources from my computer (which is streaming to my channel) to partner's laptop (which is also streaming to her channel). Using this method, we are able to share key raw feeds which can quickly be integrated with both of our overlays and audio mixing/loudness preferences.

**Comment by Riotline** [comment]
@fgolman 
With OBS 28, the groundwork for additional native auxiliary/sub AV mix was laid. The first use was to convert the virtual camera into using its own AV thread/mix rather than relying on taking the program feed thread/mix. 

What @tt2468 intends is to utilise the new groundwork to allow for actual aux/bus outputs that can have assigned inputs from scenes and audio channels. Through this, you could still route a single scene or potentially source to an aux/bus output but it would be through a separate system rather than the filter workaround.

**Comment by fgolman** [comment]
I see, that now makes better sense.

Thanks for that clarification!

Looking forward to this!



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ChicagoScanningMG** [comment]
> > With that in mind, I**'m interested to hear any notable use-cases where this would not be an effective replacement.** With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.
> 
> Potential use case. Myself and my partner co-stream. Using NDI filters, we are able to share the raw streams of several video and audio capture sources from my computer...

The NDI filters output is a killer feature.  My partner and I do something similar.  I'll stream, but we'll both be playing Dead By Daylight.  Using NDI, she can send me her raw game stream and webcam so that I can composite them into my stream.

**Comment by justibro** [comment]
> > A much better alpha build is now available: https://github.com/Palakis/obs-ndi/actions/runs/2977380826
> > The only thing it's currently technically missing is the NDI filter. I have some mixed thoughts on it, so I might as well ask the audience.
> > The NDI filter is a bit of a hack of OBS. It puts a notable strain on the main render thread, and generally does a lot of bad things that can cause problems to propagate to other parts of OBS. This mainly comes as a result of video sending.
> > My idea is to implement the aux outputs system, allowing you to manage a list of aux outputs via a centralized UI. You select your input/scene, then **select an optional audio track** to use. There is not an effective way to get the raw, unprocessed feed of an OBS source right now that also includes correct timing information. The closest thing we have is audio tracks. I personally feel this functionality would help people more than it would hurt.
> > With that in mind, I'm interested to hear any notable use-cases where this would not be an effective replacement. With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.
> 
> Hey, how can i Download it? There is no Package or am i just Blind?
> 
> Never Mind, my Browser did not Highlitet the Link XD

@DerThaller I think I'm having the same issue finding it.  Where did you find the download?

**Comment by Riotline** [comment]
@ChicagoScanningMG 
I believe that https://github.com/Palakis/obs-ndi/issues/745#issuecomment-1238799204 should apply to this as well. You should be able to do the same thing with the new system but it will be a more robust and less "hacky" solution that filters were.

**Comment by ChicagoScanningMG** [comment]
@Riotline fer sure, but is there any kind of timeline or roadmap for that to be implemented?

**Comment by DerThaller** [comment]
> 

Hey, just click on the Link (see Picture).
![grafik](https://user-images.githubusercontent.com/112657316/188958175-a292e86b-a2c7-4302-bfd4-1c1dcee2923f.png)

**Comment by OGNetheri** [comment]
> @ChicagoScanningMG I believe that [#745 (comment)](https://github.com/Palakis/obs-ndi/issues/745#issuecomment-1238799204) should apply to this as well. You should be able to do the same thing with the new system but it will be a more robust and less "hacky" solution that filters were.

Is it possible to use this to cross-source so that you can get dedicated sources also back to game-pc from stream-pc with only as a sound source or as a full video+sound source as you can do now, if need be?

**Comment by Riotline** [comment]
@ChicagoScanningMG as far as I'm aware, tt2468 has already began implementing quite a bit of the groundwork a few days ago however some work would still need to go in to it before it's implemented as functional user interface and experience.

Since obs-ndi is currently not in a "proper" release stage yet for OBS 28, parts will be changing relatively rapidly. As "alpha" may infer, the version from the workflow action artifacts are basically the MVP (minimum viable product) to be considered functional obs-ndi. 
Quite a few bugs have been quashed already while some may still be lurking. Feature parity isn't exact but still has essentials right now. You now what I mean.

Tl;dr, hopefully soon with rapid development but depends on tt2468

**Comment by Riotline** [comment]
> Is it possible to use this to cross-source so that you can get dedicated sources also back to game-pc from stream-pc with only as a sound source or as a full video+sound source as you can do now, if need be?

I don't see why this couldn't be achieved since it should drop in as a method for what you could consider routing "inputs" (the range of which whether source, scene, etc. Is unclear right now) to ndi outputs.

**Comment by Lame-One** [comment]
> > 
> 
> Hey, just click on the Link (see Picture). ![grafik](https://user-images.githubusercontent.com/112657316/188958175-a292e86b-a2c7-4302-bfd4-1c1dcee2923f.png)

It seems you need to be logged into Github for those to be clickable links, fwiw.

**Comment by Exhiibits** [comment]
> > > 
> > 
> > 
> > Hey, just click on the Link (see Picture). ![grafik](https://user-images.githubusercontent.com/112657316/188958175-a292e86b-a2c7-4302-bfd4-1c1dcee2923f.png)
> 
> It seems you need to be logged into Github for those to be clickable links, fwiw.

Using the versions above, I don't get the option to add NDI Audio/Filters to Send and receive audio from one pc to the other.

**Comment by nacalai** [comment]
It just hasn't been implemented yet. He's making a fancy graphical
interface for it which should be pretty awesome

**Comment by nacalai** [comment]
> > > > 
> > > 
> > > 
> > > Hey, just click on the Link (see Picture). ![grafik](https://user-images.githubusercontent.com/112657316/188958175-a292e86b-a2c7-4302-bfd4-1c1dcee2923f.png)
> > 
> > 
> > It seems you need to be logged into Github for those to be clickable links, fwiw.
> 
> Using the versions above, I don't get the option to add NDI Audio/Filters to Send and receive audio from one pc to the other.

It just hasn't been implemented yet. He's making a fancy graphical
interface for it which should be pretty awesome

**Comment by OGNetheri** [comment]
https://github.com/dev47apps/obs-ndi/releases/tag/4.9.2

> > > > 
> > > 
> > > 
> > > Hey, just click on the Link (see Picture). ![grafik](https://user-images.githubusercontent.com/112657316/188958175-a292e86b-a2c7-4302-bfd4-1c1dcee2923f.png)
> > 
> > 
> > It seems you need to be logged into Github for those to be clickable links, fwiw.
> 
> Using the versions above, I don't get the option to add NDI Audio/Filters to Send and receive audio from one pc to the other.

Use this while waiting what comes out of the official version: https://github.com/dev47apps/obs-ndi/releases/tag/4.9.2

**Comment by lordgemi** [comment]
> > 
> 
> Hey, just click on the Link (see Picture). ![grafik](https://user-images.githubusercontent.com/112657316/188958175-a292e86b-a2c7-4302-bfd4-1c1dcee2923f.png)

Where exactly can i find this?

**Comment by tt2468** [comment]
https://github.com/Palakis/obs-ndi/actions/runs/2977380826

It was linked above in the thread.

**Comment by jeremydk** [comment]
I spent a little time chatting with @tt2468 since I felt like there's a lot of confusion around the proposed "Aux Outputs" usage versus the current obs-ndi filters, and I wanted to hopefully help discuss some of the design such that we can be sure that the needs of the community are being fulfilled.

This is based on _my_ understanding, so I apologize if any of it comes across as more confusing than not.

Currently, in OBS, we have the ability to assign our **audio sources** to Tracks via the **Advanced Audio Properties** panel.  For streamers, this is primarily used to separate out a Twitch VOD Track from the primary Stream track.

Under the proposed design, there would be the ability to add **Aux Outputs**, each composed of an NDI Name, a Source/Scene for Video (only! Audio not included here), and selector to optionally pick the **Audio Track** to send with the video.  This allows a lot of flexibility as to how you want to group your **audio sources** prior to sending them down NDI -- as well as being able to easily customize different "views" to send down NDI.

I hope that this clears things up for folks such that we can ensure needs are being met, and that feedback is based on a common understanding.

**Comment by pvpcaleium** [comment]
@tt2468  @jeremydk First I would like to say thank you for taking the time to make NDI work in general, I don't have the skill to do so, so I appreciate it, as I am sure many of the others who's videos I have watched on how to use NDI , do also. 

What Jeremydk explained is exactly how I use them today minus the extra audio track. I am really just getting started streaming.

If I may add my usage of the Dedicated NDI outputs as an example of their benefits. Currently with my dual streaming setup I have 3 Dedicated NDI outputs. 1. Game Capture, 2. Screen Capture, 3. Desktop Audio. I use 2 scene's on my Gaming PC One for Gaming Capture and a second for Screen Capture, that I switch with a keyboard shortcut. On my streaming PC I have Two NDI Scene's, 1 for the Game and 1 for the Screen, But both have the same NDI Desktop Audio(Audio Only). 

This gives me the chance to seamlessly switch between the two video sources with the same common audio source. I can also easily, using a hot key mute the remote Desktop audio only if I need to. It also prevents me from capturing my audio twice from the remote in game audio and the local streaming mic audio. I could even take the dedicated audio and record it to its own track for who knows how many different reasons. 

What I noticed in the long run was a fair performance gain on using Dedicated NDI outputs over the NDI Output settings where it was all one massive Chunk. I believe that is because of the Audio Only setting where I can grab audio by its self and just video only as needed. By using the same NDI output audio in both places it reduced the over all CPU need. When the NDI Gaming scene was selected with the audio, it didn't need to process the NDI Display capture since it was not selected and thus uses less CPU.

Thanks again, this is just how I use NDI with OBS from limited understanding of the two overall. 
Have a wonderful day

**Comment by OGNetheri** [comment]
> I spent a little time chatting with @tt2468 since I felt like there's a lot of confusion around the proposed "Aux Outputs" usage versus the current obs-ndi filters, and I wanted to hopefully help discuss some of the design such that we can be sure that the needs of the community are being fulfilled.
> 
> This is based on _my_ understanding, so I apologize if any of it comes across as more confusing than not.
> 
> Currently, in OBS, we have the ability to assign our **audio sources** to Tracks via the **Advanced Audio Properties** panel. For streamers, this is primarily used to separate out a Twitch VOD Track from the primary Stream track.
> 
> Under the proposed design, there would be the ability to add **Aux Outputs**, each composed of an NDI Name, a Source/Scene for Video (only! Audio not included here), and selector to optionally pick the **Audio Track** to send with the video. This allows a lot of flexibility as to how you want to group your **audio sources** prior to sending them down NDI -- as well as being able to easily customize different "views" to send down NDI.
> 
> I hope that this clears things up for folks such that we can ensure needs are being met, and that feedback is based on a common understanding.

Does he told you, how is the audio in Dedicated NDI video output synced with the wanted audio track from that same source in this new system so that it doesn't need manual syncing and HOW is the audio from wanted application extracted so that it doesn't go to other channels and also stays on the same channel on the actual streaming PC? And so is it still possible to separate game, music, mic and discord for their separate audios which are also synced with the video of the source in this new method? I mean the old method (still used on the forked OBS-NDI by dev47apps) works as it should so I for one can't see why to fix something that works?

**Comment by tt2468** [comment]
> Does he told you, how is the audio in Dedicated NDI video output synced with the wanted audio track from that same source in this new system so that it doesn't need manual syncing and HOW is the audio from wanted application extracted so that it doesn't go to other channels and also stays on the same channel on the actual streaming PC? And so is it still possible to separate game, music, mic and discord for their separate audios which are also synced with the video of the source in this new method? I mean the old method (still used on the forked OBS-NDI by dev47apps) works as it should so I for one can't see why to fix something that works?

I recommend learning a bit about OBS and the track system in order to help answer your questions. Sync is not relevant here.

I think I have my answers for now. Seems people don't quite get the idea or are misunderstanding my descriptions, so I think I'll go ahead and just build it as I was planning, release it with some kind of basic intro, and let people figure it out. Using tracks is not difficult, and allows people to still do what they want within reason compared to 4.x, plus more.

**Comment by OGNetheri** [comment]
> > Does he told you, how is the audio in Dedicated NDI video output synced with the wanted audio track from that same source in this new system so that it doesn't need manual syncing and HOW is the audio from wanted application extracted so that it doesn't go to other channels and also stays on the same channel on the actual streaming PC? And so is it still possible to separate game, music, mic and discord for their separate audios which are also synced with the video of the source in this new method? I mean the old method (still used on the forked OBS-NDI by dev47apps) works as it should so I for one can't see why to fix something that works?
> 
> I recommend learning a bit about OBS and the track system in order to help answer your questions. Sync is not relevant here.
> 
> I think I have my answers for now. Seems people don't quite get the idea or are misunderstanding my descriptions, so I think I'll go ahead and just build it as I was planning, release it with some kind of basic intro, and let people figure it out. Using tracks is not difficult, and allows people to still do what they want within reason compared to 4.x, plus more.

Well for with just logicly and as a user of NDI I have noticed that the sync between audio and video IS quite important. As for me, I use sync with source option and in some case the manual sync is still needed between audio and video which has been separated but have same origin of source. That case e.g. comes when one wants to use application app audio for taking game sounds from game as a separate source and use some of  LiveVisionKit's options. Also without even using Livevisionkit,  using  main-ndi as a video source for because when game capture source is used with dedicated ndi-output with sound, it  for some reason starts to lag encoding and drop frames. Just only for the latter reason, syncing between NDI-audio&video seems a bit important and needed for these things to work. Of course if I'm wrong, then please explain, why is there syncing option in receiving devices OBS for even the audio only? 🤔

**Comment by tt2468** [comment]
> Well for with just logicly and as a user of NDI I have noticed that the sync between audio and video IS quite important. As for me, I use sync with source option and in some case the manual sync is still needed between audio and video which has been separated but have same origin of source. That case e.g. comes when one wants to use application app audio for taking game sounds from game as a separate source and use some of LiveVisionKit's options. Also without even using Livevisionkit, using main-ndi as a video source for because when game capture source is used with dedicated ndi-output with sound, it for some reason starts to lag encoding and drop frames. Just only for the latter reason, syncing between NDI-audio&video seems a bit important and needed for these things to work. Of course if I'm wrong, then please explain, why is there syncing option in receiving devices OBS for even the audio only? 🤔

Perhaps I should repeat it again. **This conversation has nothing to do with AV sync issues.** I don't know why you are bringing this up.

**Comment by Riotline** [comment]
Just so anyone wants a simple explanation of how the NDI Aux Outputs will (probably) function and how OBS currently handles audio tracks, here is a short video about it:

https://user-images.githubusercontent.com/16257629/189654913-8fb4c149-aa29-4235-baf9-1870fffbf79a.mp4

Please note that due to the plugin not in a finished state, this could change. This is to assist with any misconceptions or misunderstandings on what is changing when we mention that NDI Filters will be superseded by NDI Auxillary Outputs.

**Comment by OGNetheri** [comment]
> Just for anyone wanting a **HEAVILY SIMPLIFIED** explanation of how the NDI Aux Outputs will (probably) function and how OBS currently handles audio tracks, here is a short video about it:
> 
>  2022-09-12.22-29-05.mp4 
> Please note that due to the plugin not being in a finished state as of yet, this is subject to change. This is to assist with any misconceptions or misunderstandings on what is changing when we mention that NDI Filters will be superseded by NDI Auxillary Outputs.

Sorry for being simple and stupid but how this is better than having separate application audio sources with Dedicated NDI-filter in application audio capture (plug-in or the v28 native beta)  in which you can specify the channels used for the audio  in stream and VOD? I mean e.g. using the "application audio capture" with ndi-audio only filter and specifying vod-track in OBS settings to 6 and leaving that without enabling on stream pc's NDI inputs "advanced audio properties" input gives the result very effectively and does not save the music in this example into vods same time if needed, you can save it on recordings, jusat the same manner as in this NDI-aux? Also this same method works of course on other wanted sources too, 

I'm not being difficult just to be annoying, I just can't understand why to change something which works on V28 already with dev47app's fork of the original solution and has worked on V27 since the Palakis made OBS-NDI with the filter options and bozbez gave us the "Win capture audio"-plugin.

**Comment by krystofkrticka** [comment]
> I'm not being difficult just to be annoying, I just can't understand why to change something which works on V28 already with dev47app's fork of the original solution and has worked on V27 since the Palakis made OBS-NDI with the filter options and bozbez gave us the "Win capture audio"-plugin.

Well, dev47app's fork is kind of useless in long term since it does not support **NDI 5.5**, but it's only good as temporary solution until there is plugin which supports newest version of NDI.

**Comment by Riotline** [comment]
@OGNetheri 
So what tt2468 has been doing is rewriting the obs-ndi plugin to work with both the new qt6 standard from OBS as well as the NDI v5+. In general, the rewrite is, as the name suggests, a rewrite. 

> Sorry for being simple and stupid but how this is better than having separate application audio sources with Dedicated NDI-filter in application audio capture (plug-in or the v28 native beta) in which you can specify the channels used for the audio in stream and VOD?

They have opted out of doing the dedicated filter ndi output for a few reasons, mainly relating to the 'hacky method' that previous versions used to capture the sources/scenes. The upcoming method would utilise new native OBS interfaces to create appropriate outputs out to NDI. Of course, as you have noted, there are some pitfalls. Unfortunately, the audio will be limited to the six tracks that OBS already provides. They are using these tracks in lieu of the 'hacky' filters to avoid doing said hacky method amongst other reasons; however, using audio tracks does allow for mix-minusing natively in OBS and outputing an entire track through NDI Out.

On a separate note, in the future you will be able to output out of OBS through filters using VST3 once/if that support is added (which it is in early development/planning). This would be done by utilising NDI Audio Direct via a VST3 plugin.

**Comment by keith8496** [comment]
> With that in mind, I'm interested to hear any notable use-cases where this would not be an effective replacement. With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.

We currently use the filter to send a scene that is more-or-less a poor man's multi-view to a greenroom TV. If we could  send out the OBS multi-view window with program audio via NDI that would be great.

**Comment by OGNetheri** [comment]
> > I'm not being difficult just to be annoying, I just can't understand why to change something which works on V28 already with dev47app's fork of the original solution and has worked on V27 since the Palakis made OBS-NDI with the filter options and bozbez gave us the "Win capture audio"-plugin.
> 
> Well, dev47app's fork is kind of useless in long term since it does not support **NDI 5.5**, but it's only good as temporary solution until there is plugin which supports newest version of NDI.

Really? I installed the ndi 5 from newtek and got "ndi source 5" ndi sources in OBS. After I installed dev47apps fork, i get non-5 version of "NDI source" and the dedicated filters for it and the old sources with NDI work again as they worked in V27?  So I guess my V28 and forked OBS-NDI are somehow special as they work together as they workd on v27.#? 🤔

**Comment by OGNetheri** [comment]
> They have opted out of doing the dedicated filter ndi output for a few reasons, mainly relating to the 'hacky method' that previous versions used to capture the sources/scenes. The upcoming method would utilise new native OBS interfaces to create appropriate outputs out to NDI. Of course, as you have noted, there are some pitfalls. Unfortunately, the audio will be limited to the six tracks that OBS already provides. They are using these tracks in lieu of the 'hacky' filters to avoid doing said hacky method amongst other reasons; however, using audio tracks does allow for mix-minusing natively in OBS and outputing an entire track through NDI Out. As long as you dont need more mixes than the 6 tracks given, then everything is the same but with more functionality.
> 
Ok so am I able to make it's own "dedicated audio" with this new system for the "application audio capture" as I am able to do now? You put "use these" marks on the NDI AUX and on the receiving OBS I put "receive these channels and send them to stream/video" markings for the "NDI AUX IN"-source's properties so that I can still rule e.g. music out of recording on stream?

**Comment by krystofkrticka** [comment]
Well NewTek probably has some kind of backwards compatibility but old ndi plugin is unable to use some of new features of ndi 5.5. It uses only features which already were in ndi before version 5

**Comment by OGNetheri** [comment]
> Well NewTek probably has some kind of backwards compatibility but old ndi plugin is unable to use some of new features of ndi 5.5. It uses only features which already were in ndi before version 5

So basically as there's backwards compatibility, for my and for many others use case the current way would've been enough with just making the plug-in compatible with V28? 🤔

**Comment by krystofkrticka** [comment]
> > Well NewTek probably has some kind of backwards compatibility but old ndi plugin is unable to use some of new features of ndi 5.5. It uses only features which already were in ndi before version 5
> 
> So basically as there's backwards compatibility, for my and for many others use case the current way would've been enough with just making the plug-in compatible with V28? 🤔

Yes that fork is good to get you up and running in meantime while you wait for ndi 5 version of plugin.

**Comment by OGNetheri** [comment]
> > > Well NewTek probably has some kind of backwards compatibility but old ndi plugin is unable to use some of new features of ndi 5.5. It uses only features which already were in ndi before version 5
> > 
> > 
> > So basically as there's backwards compatibility, for my and for many others use case the current way would've been enough with just making the plug-in compatible with V28? 🤔
> 
> Yes that fork is good to get you up and running in meantime while you wait for ndi 5 version of plugin.

My point is, why should I or anyone who's using the working fork update to other version, if the usage of the fork is enough?  🤔 Unless for some reason V28 prohibits it's use in it's way to works which wasn't prohibited at least in about 2 years I've been using the original?

**Comment by nacalai** [comment]
> > With that in mind, I'm interested to hear any notable use-cases where this would not be an effective replacement. With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.
> 
> We currently use the filter to send a scene that is more-or-less a poor man's multi-view to a greenroom TV. If we could  send out the OBS multi-view window with program audio via NDI that would be great.

He's asking for reasons not to use the new method. You will of course still be able to output scenes as you wish

**Comment by krystofkrticka** [comment]
I get your point of view, but why you might want to use ndi 5 version of plugin are network communication improvements in ndi 5.

**Comment by OGNetheri** [comment]
> I get your point of view, but why you might want to use ndi 5 version of plugin are network communication improvements in ndi 5. But as I said in one of my previous comments this version of plugin is not yet ready for prime time. So that's the main reason why I also think it is good to have at least working fork of old version.

At least on NDI's recommended 1Gb LAN I've had no issues with the old method ever despite some bug which puts the source on wrong input sometimes if streampc's OBS is started before game pc's even when I have cross sourcing to that both pc's send data for each other as NDI sources. Other issue seems to be that if main-ndi is disabled and only dedicated NDI in the capture is used, for some reason OBS starts to lag. But the latter is not audio issue or LAN issue. 🤔 

If new method offers the same customability with (cross) sources with the help of "application audio capture" it's fine but I for myself can go with the forked version until it gets unsupported by dev or OBS, as for now I have no reason to update on the new method. Also I'm still wondering is the "hackyness" the only true reason to update this Palakis's original.

**Comment by aramg** [comment]
As noted in the following post, the fork is 100% intended to be stop-gap until the ongoing re-write is complete.
https://github.com/obsproject/obs-studio/discussions/7072#discussioncomment-3543748

But if the 4.x version for OBS28 works fine for your needs; you can absolutely keep using it as long as it works.
I think it will be continue to work for a while, until there is another major OBS update or a major NDI update.

Switching to use NDI-5 is going to need to happen sooner or later, and it makes sense to do it now.
And reading through the comments, seems the main reason for restructuring how outputs work is due to OBS internals.
The new approach will be more efficient (?), cleaner, and more "correct" (internally).

**Comment by OGNetheri** [comment]
Here's a reason not to remove Dedicated NDI audio output: If source is sent to other machine with Dedicated NDI output filter with or without sound enabled, "Average time to render frame" get lot's higher than with only using Dedicated NDI audio output filter. 

Even when there's no actual video to render, this Dedicated NDI ouput keeps frame time higher and if you have multiple of these sources, it stacks and eventually you get dropped frames. This also is big problem if used on Gaming PC's OBS: if one uses Dedicated NDI output filter with e.g "Game Capture", it uses much more resources than using Main NDI to move whole game, mute that NDI source on streaming PC and then use "application audio capture" with Dedicated NDI audio filter.

This I haven't ever understood, why useinfg Dedicated (video) Output filter on individual source uses much more Encoding time than "system wide" NDI ouputting?

**Comment by AyuLovelace** [comment]
Hi :) 
Just wanted to say, thank you guys for your efforts. I just used the fork to make the ndi interface work with OBS 28. :) Can't wait till the final version comes out.

**Comment by nacalai** [comment]
Will this new version possibly have the possibility to send lower bandwidth
outputs/NDIHX? I might've missed it being mentioned already so, sorry if it
has already been covered.

tor. 15. sep. 2022, 19:14 skrev Ayu Lovelace ***@***.***>:

**Comment by paulpv** [comment]
> Will this new version possibly have the possibility to send lower bandwidth outputs/NDIHX? I might've missed it being mentioned already so, sorry if it has already been covered. tor. 15. sep. 2022, 19:14 skrev Ayu Lovelace ***@***.***>:

@nacalai It means that is a possibility. @tt2468 thinks that NDIHX requires the `NDI Advanced SDK`, which has a different licensing model. I am researching NDIHX for my own [not directly related] project and will be glad to help more with this plugin once I figure that out.

If it ends up requiring a special license, perhaps NDI would be willing to provide a license for this OS project. 🤷‍♂️

**Comment by fgolman** [comment]
Really hope that can work out as HX is a great solution when you've got bandwidth limitations.

Thanks!

Fritz Golman


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by trbutler** [comment]
Are all the users reporting using the fork right now using Windows? I seem to only see Windows on the Dev47 page, not MacOS. Hopefully the official update will soon fill that gap...

**Comment by SirBryan** [comment]
Running an M1 Mac Studio with the new OBS 28. I installed the alpha build linked earlier in the thread, and it complains that the runtime isn't installed. The new.tk URL the plugin provides isn't valid, so I manually installed the latest version of NDI Tools for macOS from NDI's website.  Unfortunately the plugin still can't see the NDI 5 Runtime.

**Comment by jaitaiwan** [comment]
I just tried the same thing as SirBryan and have a similar problem where none of the NDI output settings appear, nor is there an NDI source available but there is NewTek NDI Video available in the video capture device list.

**Comment by DasKlose** [comment]
Hey guys, is there any way to use HDR Gaming (Gaming PC) SDR Streaming (Streaming PC) with the NDI Alpha Plugin?

**Comment by nacalai** [comment]
> Hey guys, is there any way to use HDR Gaming (Gaming PC) SDR Streaming (Streaming PC) with the NDI Alpha Plugin?

It has no outputs yet, so I guess that's a no, for the time being

**Comment by DasKlose** [comment]
@nacalai Thx m8! With Teleport i can play Games with HDR on my Gaming PC and send it to my Streamingpc without extrem shining. There's a little lag but its okay.

**Comment by Chaniro** [comment]
![martha](https://user-images.githubusercontent.com/114219890/191927536-5dd15e59-9642-40ca-969d-b6a6ec611242.jpg)
Hey guys, it's my first time here, i tried to download the alpha since NDI not working on the OBS V28.

it's "working" but i got this green screen, do you know where does it come from ? or to fix it ?

thank you !

**Comment by RodriMora** [comment]
> Running an M1 Mac Studio with the new OBS 28. I installed the alpha build linked earlier in the thread, and it complains that the runtime isn't installed. The new.tk URL the plugin provides isn't valid, so I manually installed the latest version of NDI Tools for macOS from NDI's website. Unfortunately the plugin says it still can't see the NDI 5 Runtime.
> 
> However, under Video Capture device, I see NDI. So perhaps something's working... I'll try it out tonight.

is the ndi website working? i tried to download it for mac here but I get a cloudflare error not finding the site https://www.ndi.tv/tools/
Is there any way to get a NDI plugin to work on version 28 for mac m1?

**Comment by nacalai** [comment]
> ![martha](https://user-images.githubusercontent.com/114219890/191927536-5dd15e59-9642-40ca-969d-b6a6ec611242.jpg)
> Hey guys, it's my first time here, i tried to download the alpha since NDI not working on the OBS V28.
> 
> it's "working" but i got this green screen, do you know where does it come from ? or to fix it ?
> 
> thank you !

You can use the Teleport plugin while we wait for ndi updates, which is great for just local pc to pc streaming (which it looks like you are doing). Or just use obs v27 with ndi

**Comment by Chaniro** [comment]
thank you :)

**Comment by Heidecker** [comment]
> ![martha](https://user-images.githubusercontent.com/114219890/191927536-5dd15e59-9642-40ca-969d-b6a6ec611242.jpg) Hey guys, it's my first time here, i tried to download the alpha since NDI not working on the OBS V28.
> 
> it's "working" but i got this green screen, do you know where does it come from ? or to fix it ?
> 
> thank you !

I was able to resolve a green screen like this in OBS by going to **Settings -> Advanced -> Video  -> Color Range** and changing this setting to **Limited**. For streaming, I have:

Color Format = NV12
Color Space = Rec. 709
Color Range = Limited

**Comment by Chaniro** [comment]
thanks !

**Comment by AlexPresso** [comment]
That new OBS tracks compatibility will be really cool, thank you very much for all the work you do :)

**Comment by OGNetheri** [comment]
@Heidecker thank you for this. It also seems that the "Limited" has to be on gaming pc's OBS only although on it's preview there's no green screen only on stream pc's OBS preview. Weird.

**Comment by gustavoninetyone** [comment]
For those who want to compile a version of the plugin translated to `Portuguese Brazil (PT-BR)` I left `pt-BR.ini` posted here

Para quem deseja compilar uma versão do plugin traduzido para o` Portugues Brasil(pt-BR)`, deixei postado aqui o` pt-BR.ini`
[https://github.com/gustavoninetyone/obs-ndi](url)

**Comment by matbox** [comment]
@tt2468  

I installed the alpha plugin on my Mac, however I do not have any Program/Preview sources available (checked via NDI Video Monitor). It works if I use the NDI Test Patters app. Is this an expected issue? 

Thanks!

**Comment by redstyle82** [comment]
> @tt2468
> 
> I installed the alpha plugin on my Mac, however I do not have any Program/Preview sources available (checked via NDI Video Monitor). I did enable them in the obs-ndi Settings.
> 
> Using NDI 5 sources in my scenes works and creating other non-OBS sources for NDI works as well on the same Mac (e.g. using the NDI Test Patters app).
> 
> Is this an expected issue?
> 
> Thanks!

hello, how can u install it? I cant find the package to install :(

**Comment by Crosskaese** [comment]
Did someone have any luck with getting the rewrite version to run with the flatpak version of OBS on Linux?
I put the libndi.so.5 lib in each dir obs-ndi searches for. Tried the 5.5 Version and the 5.0.10 Version.
Even looked at the code that loads them. Did not find a reason why it doesn't work.
Anyone any tips for me?

**Comment by Raskull13** [comment]
@tt2468 Thanks for all you're doing. So far, I'm sticking with OBS 27 so as not to take a risk on the alpha plugins. No offense meant, but right now OBS 27 works for me and I will wait it out until the OBS-NDI plugin is deemed compatible. Since I haven't read up on why OBS 28 is supposed to be so much cooler, I don't know what I'm missing. ;-) I will keep checking the OBS 28 Plugin Compatibility page to see if OBS-NDI is updated and also this thread to see if you release a Beta version that you feel is stable enough for production use.

**Comment by nacalai** [comment]
> @tt2468 Thanks for all you're doing. So far, I'm sticking with OBS 27 so as not to take a risk on the alpha plugins. No offense meant, but right now OBS 27 works for me and I will wait it out until the OBS-NDI plugin is deemed compatible. Since I haven't read up on why OBS 28 is supposed to be so much cooler, I don't know what I'm missing. ;-) I will keep checking the OBS 28 Plugin Compatibility page to see if OBS-NDI is updated and also this thread to see if you release a Beta version that you feel is stable enough for production use.

I've tested the alpha and it works wonderfully. You only have ndi source capabilities and nothing else yet..but..it works as it should. But yea..you don't have any outputs yet. Looking forward to an update!

**Comment by Raskull13** [comment]
> > @tt2468 Thanks for all you're doing. So far, I'm sticking with OBS 27 so as not to take a risk on the alpha plugins. No offense meant, but right now OBS 27 works for me and I will wait it out until the OBS-NDI plugin is deemed compatible. Since I haven't read up on why OBS 28 is supposed to be so much cooler, I don't know what I'm missing. ;-) I will keep checking the OBS 28 Plugin Compatibility page to see if OBS-NDI is updated and also this thread to see if you release a Beta version that you feel is stable enough for production use.
> 
> I've tested the alpha and it works wonderfully. You only have ndi source capabilities and nothing else yet..but..it works as it should. But yea..you don't have any outputs yet. Looking forward to an update!

I use the OBS-NDI plugin to capture 3 screens (1 with audio) from 1 computer (a virtual pinball computer) to broadcast from my streaming computer. In your opinion, will the alpha plugin handle this? I kept seeing stuff about audio in this thread and it gave me the impression that I may really want to wait. 

On the flip side, what's so hot about OBS 28 that I would want to risk my streams on an alpha plugin?

**Comment by dsoundmn** [comment]
Hi Folks - Great work so far, is there any kind of help needed? I don't see that addressed here.

Thank you!!!

**Comment by redstyle82** [comment]
> > > @tt2468 Thanks for all you're doing. So far, I'm sticking with OBS 27 so as not to take a risk on the alpha plugins. No offense meant, but right now OBS 27 works for me and I will wait it out until the OBS-NDI plugin is deemed compatible. Since I haven't read up on why OBS 28 is supposed to be so much cooler, I don't know what I'm missing. ;-) I will keep checking the OBS 28 Plugin Compatibility page to see if OBS-NDI is updated and also this thread to see if you release a Beta version that you feel is stable enough for production use.
> > 
> > 
> > I've tested the alpha and it works wonderfully. You only have ndi source capabilities and nothing else yet..but..it works as it should. But yea..you don't have any outputs yet. Looking forward to an update!
> 
> I use the OBS-NDI plugin to capture 3 screens (1 with audio) from 1 computer (a virtual pinball computer) to broadcast from my streaming computer. In your opinion, will the alpha plugin handle this? I kept seeing stuff about audio in this thread and it gave me the impression that I may really want to wait.
> 
> On the flip side, what's so hot about OBS 28 that I would want to risk my streams on an alpha plugin?

i've tried something similar with 2 pc, sometimes at random time the video start to lag hardly (in the old 27.3 all goes well)

**Comment by mcraee** [comment]
Where do I extract this v5 folder to? cant get it to work at all

**Comment by Zacharybinx34** [comment]
Any updates coming soon for the mac version of the plugin?

**Comment by jsp196** [comment]
Will this work on OBS27 as well?  We use NDI only for source ingestion into OBS and have run into issues where the video streams are not automatically discovered (though it helps if they exist before OBS is launched, that's not a guarantee).  Hoping that the rewrite will address some of that just because it's updated.

**Comment by APz** [comment]
Does the alpha version need some specific version of NDI runtime? I'm scratching my head about getting the plugin to load, I have the current NDI 5 bundle installed, current OBS (28.0.3) and the pkg of the alpha mentioned earlier in the comments.

**Comment by jaitaiwan** [comment]
@APz how are you running it? I've got the alpha version working well. I noticed that I have to reinstall the alpha every time I do an update, but it's been fine otherwise.

**Comment by APz** [comment]
@jaitaiwan I'm running Intel version and I took the Intel version (over universal). The installer put it in the right place (/Library/Application Support/obs-studio/plugins/) but OBS completely ignores it. Starting OBS from the shell doesn't seem to produce any errors related to the plugin, it just gets ignored. NDI 5 in itself seem to work fine with the software that came with the bundle.

**Comment by jaitaiwan** [comment]
Ahhh I'm using windows, so I'm no help sorry

**Comment by nacalai** [comment]
### A headsup. I wasn't aware of this myself; 
So, this is the dev branch for NDI 5. I thought I had to wait until there's been proper development before I could ever think to upgrade to obs v28.

There is a v28-version of  the NDI plugin (4.9) working just fine with ndi output, inputs, etc as it had in v27.  I was randomly browsing the obs-ndi on obsproject website, where it clearly in big red text says that there exists a v28 version which is here: https://github.com/dev47apps/obs-ndi/releases

Just a small heads up for people who still want to use NDI in full production without having to wait for this ndi 5 rewrite (which of course will be amazing, I am sure).

**Comment by McCraic** [comment]
Is there a Mac version of this or only a Windows version?

**Comment by trbutler** [comment]
I haven't tried either of them yet, but there are two Mac-compatible forks for the 4.9.x plugin: https://github.com/ddrboxman/obs-ndi/ and https://github.com/kilinbox/obs-ndi/ . The former from @DDRBoxman shows up as a pull request on this repository, too. They look promising!

**Comment by tt2468** [comment]
The DDRBoxman binaries are recommended to be used right now. Trying to get around to making an official release using them.

**Comment by Zacharybinx34** [comment]
> The DDRBoxman binaries are recommended to be used right now. Trying to get around to making an official release using them.

What's the difference? And what do you mean "trying to get around making an official release using them?"

Thanks!

**Comment by schrauger** [comment]
> A much better alpha build is now available: https://github.com/Palakis/obs-ndi/actions/runs/2977380826
> 
> The only thing it's currently technically missing is the NDI filter. I have some mixed thoughts on it, so I might as well ask the audience.
> 
> The NDI filter is a bit of a hack of OBS. It puts a notable strain on the main render thread, and generally does a lot of bad things that can cause problems to propagate to other parts of OBS. This mainly comes as a result of video sending.
> 
> My idea is to implement the aux outputs system, allowing you to manage a list of aux outputs via a centralized UI. You select your input/scene, then **select an optional audio track** to use. There is not an effective way to get the raw, unprocessed feed of an OBS source right now that also includes correct timing information. The closest thing we have is audio tracks. I personally feel this functionality would help people more than it would hurt.
> 
> With that in mind, I'm interested to hear any notable use-cases where this would not be an effective replacement. With adequate reason, I could reimplement the NDI filter, but I would very likely only add audio output, and skip video output altogether.

Note to anyone who wants to download these packages: You have to login to a GitHub account in order to have clickable links. If you view that page as an anonymous user, the Artifacts section doesn't have links to download the packages. Once logged into GitHub, they'll become links.

**Comment by terjenm** [comment]
Hi

Not sure if this is the right place, but seems like this thread has become a general OBS 28 NDI plugin issue thread, so I take my chances and add to this thread.

I have tried https://github.com/Palakis/obs-ndi/actions/runs/2977380826 (downloaded todays build) and https://github.com/dev47apps/obs-ndi/releases (for a few weeks)

Both seems to be stable in my setup.

However, I see a 200-300 ms lag between what OBS shows on the screen and what NDI tools 5 Video Monitor see on a different computer. Same switch, no CPU problem on either computer. Tried this in 2 different setups.

If I output on a decklink card and capture through an elgato camlink 4k, I see 40-100ms delay in comparison.

I am not 100% sure, but I think the delay between OBS and NDI Video Monitor was close to nothing (sub 1 frame at 60 fps) before I upgraded to OBS 28. (currently on OBS 28.1, but similar issue on earlier releases). That is, I believe going from OBS 27 to a client computer with NDI Tools had less latency than decklink -> camlink, but I may be dreaming. I am pretty sure the lag was not as much as it is now however.

The flow here is 
- "ipad showing stopwatch with hundreds of a second resolution"
- Sony a7 camera connected to PC with elgato camlink 4k. 1080p/60fps.
- Windows 11 PC with a 16 core/32 thread I9 9960x CPU, RTX 3090,  OBS 28. 1080p/60fp canvas
  Stats view on OBS shows 4% CPU, 0 frames missed or skipped, 1.5-1.7ms average time to render frame.
  
- M1 macbook pro with cabled ethernet
    - If I view output from OBS NDI with NDI video monitor 250-300ms delay (it is actually not 100% stable between tries)
    - If I use camlink 4k and grab output from a decklink card on the PC instead, 40-100ms vs. OBS screen.
    CPU is 75% idle.
    No indication of slowness from high load anywhere, lots of resources.

- I record the OBS screen and macbook screen with an iphone in slow motion video recording mode and check differences in the clock shown on the screens.

Any ideas for tuning of settings/optimizations/etc that can be done or are we potentially looking at a bug of some sort?

**Comment by terjenm** [comment]
> Hi
> 
> Not sure if this is the right place, but seems like this thread has become a general OBS 28 NDI plugin issue thread, so I take my chances and add to this thread.
> 
> I have tried https://github.com/Palakis/obs-ndi/actions/runs/2977380826 (downloaded todays build) and https://github.com/dev47apps/obs-ndi/releases (for a few weeks)
> 
> Both seems to be stable in my setup.
> 
> However, I see a 200-300 ms lag between what OBS shows on the screen and what NDI tools 5 Video Monitor see on a different computer. Same switch, no CPU problem on either computer. Tried this in 2 different setups.

After a bit more testing and digging in my own memory banks, I realized I am both dreaming and not.
NDI tools video monitor (at least on mac, but probably on windows too) adds a lot of latency.

I rolled back to a clean OBS 27 install with NDI, nothing else.
Using NDI Virtual input and zoom, I got ~40ms latency on the remote computer
Using NDI Video Monitor, 200ms latency.

OBS 28 setup seems to be 1-2 frames extra latency to that so it seems to be a tiny bit slower.
Main confusion for me is however the NDI Video Monitor. Not sure if that has gotten slower or I always tested using some app on top of Virtual Input (same app as NDI webcam if you are on windows)

Out of curiosity, I also tried the new ios/osx ventura "iphone as a webcam" functionality.
Over wifi it has about the same latency to zoom display as OSB has on the windows computer with a camlink USB adapter.

Camlink also adds about 40-60ms latency so very similar to the NDI implementation in OBS 27 and when using a birddog flex In (hdmi -> NDI) converter. This was my other mis-memorized point. The birddog flex actually has a tiny bit less (0-1 frame) lower latency than the camlink USB adapter if used as a input source for OBS 27 (I did not test on OBS 28 yet)

Still curious if there may be ways to reduce the latency more somehow?
I would think from a performance perspective, there is no reason a computer on a fixed ethernet cable could not achieve even better latency?

**Comment by APz** [comment]
Are the binaries for the alpha build still available somewhere? They appear to have expired in Github.

**Comment by jaitaiwan** [comment]
I noticed that it's expired too. I'm going to see if I can do my own workflow run to get the artifacts. If so I'll edit this comment and post the links until Palakis is able to update it.

**Comment by Zacharybinx34** [comment]
> I noticed that it's expired too. I'm going to see if I can do my own workflow run to get the artifacts. If so I'll edit this comment and post the links until Palakis is able to update it.
> 
> UPDATE: Here's the release files I generated, make sure you check to see if there's been a new tag of the rewrite branch before you look at mine in case I forget about this in the next couple months: https://github.com/jaitaiwan/obs-ndi/releases/tag/pre-1

I wouldn't count on it. Last I heard he hit a dead end and had given up on the project.

**Comment by APz** [comment]
@jaitaiwan Thank you, that did the trick. I never got the 4.x series of the plugin to get multiple camera streams to sync up well enough, in my early tests the NDI5 version did exactly that right out of the box. My setup has 3 cameras and it looks a bit demented when there's non-uniform lag between them.

**Comment by jaitaiwan** [comment]
> > I noticed that it's expired too. I'm going to see if I can do my own workflow run to get the artifacts. If so I'll edit this comment and post the links until Palakis is able to update it.
> > UPDATE: Here's the release files I generated, make sure you check to see if there's been a new tag of the rewrite branch before you look at mine in case I forget about this in the next couple months: https://github.com/jaitaiwan/obs-ndi/releases/tag/pre-1
> 
> I wouldn't count on it. Last I heard he hit a dead end and had given up on the project.

I hope not, the plugin as it stands has been working flawlessly for me in 28.0.1 with NDI 5.5.

**Comment by terjenm** [comment]
The current version seems to work quite well?

I have not seen any issues except that it sometimes takes quite a bit of time before the stream "shows up".
Once the stream is active, its been quite stable for me, although I suspect there is 1-2 frames more lag than before.

Could it simply be released to make life easier for everybody?

**Comment by APz** [comment]
> I hope not, the plugin as it stands has been working flawlessly for me in 28.0.1 with NDI 5.5.

Just out of curiosity, is this forked from the exact spot where the alpha that was been recommended had been compiled? I'm debugging what's the difference between that and the Mac version of the previous one. With the same cameras, the Mac version has normal sound, this one has garbled and clicking.

**Comment by jaitaiwan** [comment]
Sorry for the late reply @APz, it's forked from the last commit on the rewrite branch. For future reference you can actually find the answer to your question using https://github.com/Palakis/obs-ndi/network

I'm not sure why you'd have garbled and clicking issues

**Comment by MUSE-iC** [comment]
So is the rewrite/update still in the works?

**Comment by timokorkalainen** [comment]
NDI is such an important part of the OBS ecosystem for any more complex production, that one would hope for an official support by the project or by NewTek themselves. Hoping for the best!

**Comment by getify** [comment]
OBS 29 is out now... is that going to help or further complicate this effort?

**Comment by nacalai** [comment]
@MUSE-iC @timokorkalainen @getify - Just to clear any confusion. A **rewrite** is in the works. But the **normal** usual plugin has been compatible with obs v28+ a long time, using this link: https://github.com/Palakis/obs-ndi/releases/tag/dummy-tag-4.10.0

Excited for a future rewrite though!

**Comment by paulpv** [comment]
@nacalai NO!
Do **NOT** share or use that https://github.com/Palakis/obs-ndi/releases/tag/dummy-tag-4.10.0 4.10.0 release link!
That is the version that has the bug that this whole issue is about.

Use this version instead:
https://github.com/DDRBoxman/obs-ndi/releases/tag/4.10.1

In @tt2468 's own words (above), this repo's 4.10.0 release **IS** DDRBoxman's code.
Since 4.10.0, DDRBoxman patched his to fix this issue and released it in his repo as 4.10.1.
This official repo has not been updated with the patch (even though it is in a PR).
Use DDRBoxman's 4.10.1 release.

**Comment by paulpv** [comment]
@getify
> OBS 29 is out now... is that going to help or further complicate this effort?
No

**Comment by paulpv** [comment]
My bad. My reaction was because I thought I was replying to #785 
It still stands that you should not be using this repo's 4.10.0.
You should be using DDRBoxman's 4.10.1.

**Comment by paulpv** [comment]
FYI, I detail the confusion of these builds at https://github.com/Palakis/obs-ndi/issues/785#issuecomment-1383287294

**Comment by nacalai** [comment]
@paulpv Right, I am normally pretty into what happens in this github page but the audio distortion one have gone past my head. That totally explains some audio crackling issues we've been having. Going to download that build on all our machines. Thanks for pointing that out. Totally agree this should've been added to this build a long time ago. You seem like a proper man to add to the team indeed. 

Sorry to add confusion! Will edit my original comment.

**Comment by paulpv** [comment]
Try this in the latest `develop` branch `4.11.0-RC` tagged version that is due to officially release once approved: https://github.com/Palakis/obs-ndi/pull/849

**Comment by paulpv** [comment]
This seems fixed in 4.11.0-RC; Closing.
Feel free to comment if you test and disagree.
